lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb

```
@@ -0,0 +1,132 @@
+          raise
+        end
+
+        apk_output_file_path = params[:apk_output_file_path]
+        keystore_path = params[:keystore_path]
+        keystore_password = params[:keystore_password]
+        keystore_key_alias = params[:keystore_key_alias]
+        signing_key_password = params[:signing_key_password]
+
+        sh(generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password))
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.details
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :aab_file_path,
+            env_name: 'ANDROID_AAB_FILE_PATH',
+            description: 'The path to the AAB file. If not speicified, the action will attempt to read from the lane context using the `SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS` and `SharedValues::GRADLE_AAB_OUTPUT_PATH` keys',
+            type: String,
+            optional: true,
+            default_value: nil,
+            verify_block: proc { |p| UI.user_error!("AAB path `#{p}` is not a valid file path.") unless File.file?(p) }
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :apk_output_file_path,
+            env_name: 'ANDROID_APK_OUTPUT_PATH',
+            description: 'The output path where the APK file will be generated. The directory will be created if it does not yet exist',
+            type: String,
+            optional: false,
+            default_value: nil
+          ),
```

Now that we decided to not create intermediate folders if they don't exist, we should add a check to ensure the parent dir exists. Otherwise we'd still have the action run `bundletool` but only fail afterwards while trying to move the file to the destination, instead of failing early.

```suggestion
            verify_block: proc { |p| UI.user_error!('The parent folder for the destination does not exist. Please create it first.') unless p.nil? || File.directory?(File.dirname(p)) }
          ),
```